feat: first release, move tests FAILS

theiari opened the pull request:
It seems there're some issues regarding linking between rust and move. Not entirely sure on how should I properly test the move native, since `iota move test` fails.

I tried following mainly `tx_context` flow to add:

- move module `deterministic_object_id.move`
- move native `derive_id_with_salt` 
- rust function `derive_id_with_salt` someway similar to`[`derive_id`](https://github.com/iotaledger/iota/blob/develop/crates/iota-types/src/base_types.rs#L1301)

Ahead of this current commit, I also tried

- Adding a few missing lines to `iota-framework` on [lib.rs](https://github.com/iotaledger/iota/blob/develop/crates/iota-framework/src/lib.rs) related to the native function.
- Followed this README: https://github.com/iotaledger/iota/tree/d17b7f77ee7d3217a4284db24259301c4110f716/crates/iota-framework, even though `all_natives` does not exist.
- Just in case, I also tried running `update_all_snapshots.sh` in [`iota/scripts/`](https://github.com/iotaledger/iota/blob/develop/scripts/update_all_snapshots.sh).

However, when I try to run `iota move test`, around **300 out of 500 tests** fail with the error:

UNEXPECTED_VERIFIER_ERROR (code 2017) and some reference to the native function I am trying to add.

theiari committed:
feat: first release, move tests FAILS

Dkwcs commented:
We have to rebuild the `iota` crate and run move test based on it(just `cargo build` will be enough).
If we run `move test` with the new iota binary the tests will pass.

Then if we run `UPDATE=1 cargo insta test` we have some errors:

```
   │  
37 │ ╭ /// Create an `address` that has not been used. As it is an object address, it will never
38 │ │ /// occur as the address for a user.
   │ ╰────────────────────────────────────^ Documentation comment cannot be matched to a language item
```
   
```
38 │ ╭ /// Create an `address` that has not been used. As it is an object address, it will never
39 │ │ /// occur as the address for a user.
40 │ │ /// In other words, the generated address is a globally unique object ID.
   │ ─────────────────────────────────^ Documentation comment cannot be matched to a language item
```

I guess we can handle it somehow, just to know how to write Move comment correctly

theiari committed:
test: implemented few tests

fix: added missing references to native function

feat: changed native function visibility to public, simplifying the whole process

fix: removed fresh_object_address function from tx_context

theiari commented:
Not sure about `native fun derive_id_with_salt(...)`visibility. 

Right now it's `public` but maybe it's better to have it `private`?

lzpap commented on crates/iota-framework/packages/iota-framework/sources/object.move:
```diff
@@ -176,6 +180,14 @@ public fun new(ctx: &mut TxContext): UID {
 
+/// Create a new object with a precomputed objectId.
+/// Using a fixed 123 to avoid collisions
+public fun new_precomputed(iota_address: address, salt: vector<u8>): UID {
```
this function creates an UID, therefore it needs to ensure that each ID is actually unique. For the `new` function above, uniqueness is ensured via the tx context, for accounts, we need to gate access to this function via the account creation flow.

Therefore, this function should be public(package), such that it canbe called from the (future) account framework module but not by any outsider package.

lzpap replied:
+1: in the account creation flow we need to check a global table of existing accounts
